src/python/pants/base/workunit.py

```diff
@@ -189,11 +190,16 @@ def output(self, name):
       raise Exception('Invalid output name: {}'.format(name))
     if name not in self._outputs:
       workunit_name = re.sub(r'\W', '_', self.name)
-      path = os.path.join(self.run_info_dir,
-                          'tool_outputs', '{workunit_name}-{id}.{output_name}'
-                          .format(workunit_name=workunit_name,
-                                  id=self.id,
-                                  output_name=name))
+      workunit_slug = '{workunit_name}-{id}.{output_name}'.format(
+          workunit_name=workunit_name,
+          id=self.id,
+          output_name=name
+        )
+      path = os.path.join(
+          self.run_info_dir,
+          'tool_outputs',
+          sha1(workunit_slug.encode('UTF-8')).hexdigest()
```

Comment:
In Py3, we no longer need to specify the encoding for `"my_str".encode()`. It defaults to utf-8.